pkg/monitoring/metrics/forklift-controller/migration_metrics.go

```diff
@@ -26,85 +29,91 @@ func RecordMigrationMetrics(c client.Client) {
 				continue
 			}
 
-			// Holding counter vars used to make gauge update "atomic"
-			var succeededRHV, succeededOCP, succeededOVA, succeededVsphere, succeededOpenstack float64
-			var failedRHV, failedOCP, failedOVA, failedVsphere, failedOpenstack float64
+			// Initialize or reset the counter map at the beginning of each iteration
+			counterMap := make(map[string]float64)
```

Comment:
We need to fetch the current state and save it. Using the previous implementation, we ignored plans/migrations that switched different statuses and didn't count them. Later, this will be processed in telemetry for what we need. Since there are too many combinations, adding a specific counter for each type based on providerType/migrationType/destination/status, I've added a map that will hold the different combinations as keys and the counter for each as values. This map will be reset each iteration to represent the actual state of the system.